### ISSUE TITLE
Empty mapping.txt files fail builds

### ISSUE THREAD

**Issue by jschear** [bugsnag/bugsnag-android-gradle-plugin#178]
Hi! While upgrading to the latest version of the gradle plugin from version 4.0.0, I started seeing errors like this for each minified build.
```
> Task :app:uploadBugsnag<flavor>-<abi>-<buildType>Mapping FAILED
Attempting upload of mapping file to Bugsnag
Bugsnag upload failed with code 422: Received empty file
```

It seems that when proguard is configured not to obfuscate code (i.e. the`-dontobfuscate` option is present in the config file), it generates an empty `mapping.txt` file, and the Bugsnag API returns a 422 when an empty file is uploaded.

I can work around this by turning off `autoUpload` or `failOnUploadError`, but this feels like a bug to me because it's valid for proguard to emit an empty mapping file.

### Expected behavior
The build completes successfully.

### Observed behavior
The `uploadBugsnagReleaseMapping` task fails.

### Steps to reproduce
- Set up an android project with the bugsnag gradle plugin applied
- Turn on minification with proguard for a build type (say, `release`), and add a `proguardFile` containing `-dontobfuscate`
- Run `./gradlew assembleRelease`

### Version
Bugsnag gradle plugin: `4.7.0`
Bugsnag client library: `4.21.1`
Android Gradle Plugin version: `3.5.1`

### Additional information
I'm not sure if this happens when using R8 instead of proguard.

**Comment by mattdyoung** [comment]
Hi @jschear 
Thanks for the report. Like you say, it can be worked around by turning off `autoUpload` but I agree, we should be able to just ignore empty mapping files to avoid this from happening.

The reason you only saw this after the upgrade is that we only introduced the improvement to fail the build on upload errors in [v4.1.0](https://github.com/bugsnag/bugsnag-android-gradle-plugin/releases/tag/v4.1.0).

**Comment by mattdyoung** [comment]
Fixed in [v4.7.1](https://github.com/bugsnag/bugsnag-android-gradle-plugin/releases/tag/v4.7.1)
